src/test/java/parser/JsonParserTest.java

```diff
@@ -0,0 +1,108 @@
+
+@Test
+public class JsonParserTest {
+
+    private static final double BOOK_PRICE = 25.2;
+    private static final double SOFT_PRICE = 99.0;
+    private static final String PATH_TO_CART_LIST = "/Users/dmitrymandrik/Documents/training/testng-maven/src/test/java/datasets/carts.txt";
+    private static final String PATH_TO_CART = "/Users/dmitrymandrik/Documents/training/testng-maven/src/main/resources/john-cart.json";
+    private Cart cart;
+    private JsonParser parser;
+
+    @DataProvider
+    private Object[][] setCartFilesList() {
+        BufferedReader in;
+        List<String> cartFilesList = new ArrayList<>();
+
+        try {
+            in = new BufferedReader(new FileReader(PATH_TO_CART_LIST));
+            String str;
+
+            while ((str = in.readLine()) != null) {
+                cartFilesList.add(str);
+            }
+            in.close();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        Object[][] returnValue = new Object[cartFilesList.size()][1];
+        int index = 0;
+        for (Object[] each : returnValue) {
+            each[0] = cartFilesList.get(index++);
+        }
+        return returnValue;
+    }
+
+    private Cart addItemsToCartTest() {
+
+        RealItem rItem = new RealItem();
+        rItem.setName("Book");
+        rItem.setWeight(1.5);
+        rItem.setPrice(BOOK_PRICE);
+
+        VirtualItem vItem = new VirtualItem();
+        vItem.setName("IntellIJ Idea");
+        vItem.setSizeOnDisk(512.42);
+        vItem.setPrice(SOFT_PRICE);
+
+        Cart cart = new Cart("john-cart");
+        cart.addRealItem(rItem);
+        cart.addVirtualItem(vItem);
+        return cart;
+    }
+
+    @BeforeClass
+    void initialize() {
+        this.cart = addItemsToCartTest();
+        this.parser = new JsonParser();
+        parser.writeToFile(cart);
+    }
+
+    @Test(expectedExceptions = NoSuchFileException.class, dataProvider = "setCartFilesList")
+    void readFromFileTest(String file) {
+        parser.readFromFile(new File(file));
+    }
+
+    @Test
+    void validateCorrectCartNameWrittenToFileTest() {
+        Cart johnCart = parser.readFromFile(new File(PATH_TO_CART));
+        String expectedResultName = cart.getCartName();
+        String actualResultName = johnCart.getCartName();
+        Assert.assertEquals(expectedResultName, actualResultName);
+    }
+
+    @Test
+    void validateCorrectCartPriceWrittenToFileTest() {
+        Cart johnCart = parser.readFromFile(new File(PATH_TO_CART));
+        double expectedResultName = cart.getTotalPrice();
+        double actualResultName = johnCart.getTotalPrice();
+        Assert.assertEquals(expectedResultName, actualResultName);
+    }
+
+    @Test
+    void validateCorrectRealItemWrittenToFileTest() {
+        Cart johnCart = parser.readFromFile(new File(PATH_TO_CART));
+        List<RealItem> expectedResultName = cart.getRealItems();
+        List<RealItem> actualResultName = johnCart.getRealItems();
```

Comment:
Just to make sure that we are on the same page: Assert class has method to assert arrays, but as far as RealItem class is *not* overriding equals(...) method, we can't to make assertion of arrays here. Your cart contains 1 real item, so your test makes sense.

Reply:
Yes, we are on the same page. That's what I meant.